lib/workers/repository/onboarding/branch/index.ts

```diff
@@ -81,9 +78,11 @@ export async function checkOnboardingBranch(
     }
   }
   if (!GlobalConfig.get('dryRun')) {
-    logger.debug('Checkout onboarding branch.');
     // TODO #7154
-    await checkoutBranch(onboardingBranch!);
+    if (!isConflicted) {
+      logger.debug('Merge onboarding branch in default branch');
+      await mergeBranch(onboardingBranch!, true);
+    }
```

Comment:
I'm not sure why we skipped checkout out the onboarding branch if dry run, but I'm not sure why we'd need to skip this step either

Reply:
Was added here:
https://github.com/renovatebot/renovate/pull/3333
Will see if this error stil occurs and update if not.

Reply:
Error still occurs so keeping the if-check.